pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/model/event/CreateKitEventAction.java

```diff
@@ -0,0 +1,129 @@
+    }
+
+    @Override
+    public void doAction(PexInterpreter pexInterpreter, Handle handle, EventSignal signal) {
+        Integer delayBeforePosting = eventConfiguration.getPostDelaySeconds();
+        if (delayBeforePosting != null && delayBeforePosting > 0) {
+            if (!eventConfiguration.dispatchToHousekeeping()) {
+                throw new DDPException("Incompatible event configuration:"
+                        + " delayed Kit Creation events should set dispatchToHousekeeping");
+            }
+            long queuedEventId = queueDelayedEvent(handle, signal);
+            log.info("Queued Create Kit event with id {}", queuedEventId);
+        } else {
+            doActionSynchronously(handle, signal);
+        }
+    }
+
+    public void doActionSynchronously(Handle handle, EventSignal signal) {
+
+        List<KitConfigurationDto> kitConfigs = handle.attach(KitConfigurationDao.class)
+                .getKitConfigurationDtosByStudyId(signal.getStudyId());
+
+        //load user and address
+        DsmAddressValidationStatus statusType = null;
+        User user = handle.attach(UserDao.class).findUserByGuid(signal.getParticipantGuid()).get();
+        MailAddress defaultAddress = handle.attach(JdbiMailAddress.class)
+                .findDefaultAddressForParticipant(signal.getParticipantGuid())
+                .orElse(null);
+
+        if (defaultAddress == null) {
+            log.warn("Participant {} is missing a default mailing address. Deleting the create kit event", signal.getParticipantGuid());
+            return;
+        }
+
+        if (defaultAddress.getValidationStatus() == DSM_INVALID_ADDRESS_STATUS.getCode()) {
+            log.warn("Participant {} has an invalid mailing address", signal.getParticipantGuid());
+            return;
+        } else {
+            try {
+                statusType = DsmAddressValidationStatus.getByCode(defaultAddress.getValidationStatus());
+            } catch (Exception e) {
+                log.warn(e.getMessage() + ". Participant: {} ", signal.getParticipantGuid());
+                return;
+            }
+        }
+
+        Optional<KitConfigurationDto> kitConfigByTypeOpt = kitConfigs.stream().filter(dto -> dto.getKitTypeId()
+                == kitTypeId).findFirst();
+
+        KitConfiguration kitConfig = kitConfigByTypeOpt.isPresent()
+                ? handle.attach(KitConfigurationDao.class).getKitConfigurationForDto(kitConfigByTypeOpt.get()) : null;
+
+        KitCheckService service = new KitCheckService();
+        KitCheckService.PotentialRecipient candidate = new KitCheckService.PotentialRecipient(user.getId(),
+                user.getGuid(), defaultAddress.getId(), statusType);
+        KitCheckService.KitCheckResult kitCheckResult = new KitCheckService.KitCheckResult();
+        kitCheckResult = service.processPotentialKitRecipient(signal.getStudyGuid(), kitTypeId,
+                kitCheckResult, kitConfig, candidate, true);
```

Comment:
`kitConfig` may be `null` here, and `processPotentialKitRecipient(String, long, KitCheckResult, KitConfiguration, PotentialRecipient)` does not properly handle `null` values of the `KitConfiguration` parameter